lib/code_fund_web/controllers/ad_serve_controller.ex

```diff
@@ -15,31 +15,39 @@ defmodule CodeFundWeb.AdServeController do
     cond do
       template == nil ->
         templates = Templates.list_templates()
-        template_slugs = Enum.map(templates, fn c -> c.slug end)
+
+        available_templates =
+          Enum.map(templates, fn c -> c.slug end)
+          |> Enum.join("|")
 
         conn
         |> put_status(:not_found)
         |> put_resp_content_type("application/javascript")
         |> text(
-          "console.log('CodeFund template does not exist. Available templates are \\'#{
-            Enum.join(template_slugs, "','")
-          }\\'');"
+          "console.log('CodeFund template does not exist. Available templates are [#{
+            available_templates
+          }]');"
         )
 
       true ->
         theme = Enum.find(template.themes, fn t -> t.slug == theme_slug end)
 
         cond do
           theme == nil ->
-            themes = Themes.list_themes()
+            themes = Themes.list_themes_for_template(template)
             theme_slugs = Enum.map(themes, fn c -> c.slug end)
 
+            available_slugs =
+              theme_slugs
+              |> Enum.uniq()
```

Comment:
If you do a `distinct` in your sql, you won't need to enum over this list again to get uniq values, i think.

Reply:
Slugs can have the same name as long as the template is different. This is a feature. For example, multiple templates can have the "light" theme

Reply:
Ah.. i misread your comment.. good idea

Reply:
Now that i think of it, the other refactor I made makes it so there are no duplicates

Reply:
Sure, but we're not doing anything else w/ the theme list other than returning a list of uniq names so it seems like it'd be more performant to do a distinct on the field and just select `slugs` from the query rather than map over a list.